hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientInvocation.java

```diff
@@ -182,6 +184,11 @@ public void notifyException(Throwable exception) {
             return;
         }
 
+        if (address != null && exception instanceof TargetNotMemberException) {
```

Comment:
This will prevent all invocations which receive TargetNotMemberException being not retried even when it is actually retryable and it was being retried before. E.g. Imap.get will end up at invocationService.invokeOnTarget where address != null. Don't we want to retry these invocations?